charts/consul/templates/connect-inject-configmap.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if .Values.connectInject.enabled }}
```

Comment:
do we need to create this is `imagePullSecrets` is not set?

Reply:
Good call, I also need to gate the `config-file` arg passed to the connect injector `Deployment`

Reply:
It makes sense to gate this for now; however, in the future I am expecting this to be the place we put structured config values like this one that don't fit neatly into a flag value

Reply:
nvm, I see how we handle if the image pull secrets is empty in the code, always having it results in nicer code paths